UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -434,6 +442,84 @@ void UpdateSentBrainParameters(UnityRLInitializationOutputProto output)
 
         #endregion
 
+
+        #region Handling side channels
+
+        public void RegisterSideChannel(SideChannel sideChannel)
+        {
+            if (m_SideChannelsDict.ContainsKey(sideChannel.ChannelType()))
+            {
+                throw new UnityAgentsException(string.Format(
+                "A side channel with type index {} is already registered. You cannot register multiple " +
+                "side channels of the same type."));
+            }
+            m_SideChannelsDict.Add(sideChannel.ChannelType(), sideChannel);
+        }
+
+        public static byte[] GetSideChannelMessage(Dictionary<int, SideChannel> sideChannels)
+        {
+            using (var memStream = new MemoryStream())
+            {
+                using (var binaryWriter = new BinaryWriter(memStream))
+                {
+                    foreach (var sideChannel in sideChannels.Values)
+                    {
+                        var messageList = sideChannel.MessageQueue;
+                        foreach (var message in messageList)
+                        {
+                            binaryWriter.Write(sideChannel.ChannelType());
```

Comment:
Do we need to be explicit about the endianness either here or in python? See also https://docs.python.org/3/library/struct.html#byte-order-size-and-alignment
From digging, it looks like C# always write in little-endian format, but python defaults to native byte order.

Reply:
This is old, so not sure if it's still accurate: https://stackoverflow.com/questions/9618608/is-net-binaryreader-always-little-endian-even-on-big-endian-systems